### PR TITLE
Implement the `diagnostics` operator

### PR DESCRIPTION
**Pull request by Dakostu**
This change implements a new `diagnostics` operator that shows all diagnostics that a Tenzir node encountered during its runtime.

### CODE REVIEW

File: libtenzir/builtins/operators/diagnostics.cpp [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+
+  auto parse_operator(parser_interface& p) const -> operator_ptr override {
+    auto parser = argument_parser{"diagnostics", "https://docs.tenzir.com/next/"
+                                                 "operators/diagnostics"};
+    bool live = false;
+    parser.add("--live", live);
+    parser.parse(p);
+    const auto definition = fmt::format("export --internal {}| where #schema "
+                                        "== /tenzir\\.diagnostic\\.*/",
+                                        live ? " --live" : "");
```
Comment (dominiklohmann) [inline libtenzir/builtins/operators/diagnostics.cpp:37]:
```suggestion
    const auto definition = fmt::format("export --internal {}| where #schema "
                                        "== \"tenzir.diagnostic\"",
                                        live ? " --live" : "");
```

File: web/docs/operators/diagnostics.md [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+Contains detailed information about the diagnostic.
+
+|Field|Type|Description|
+|:-|:-|:-|
+|`pipeline_id`|`string`|The ID of the pipeline that received the diagnostic.|
+|`timestamp`|`time`|The exact timestamp of the diagnostic creation.|
+|`message`|`string`|The diagnostic message.|
+|`severity`|`string`|The diagnostic severity.|
+|`notes`|`list<record>`|The diagnostic notes. Can be empty.|
+|`annotations`|`list<record>`|The diagnostic annotations. Can be empty.|
```
Comment (dominiklohmann) [inline web/docs/operators/diagnostics.md:42]: Please document the records in separate tables right below like we do for operator metrics.

File: web/docs/operators/diagnostics.md [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+
+Tenzir emits diagnostic information with the following schema:
+
+### `tenzir.diagnostic`
+
+Contains detailed information about the diagnostic.
+
+|Field|Type|Description|
+|:-|:-|:-|
+|`pipeline_id`|`string`|The ID of the pipeline that received the diagnostic.|
```
Comment (dominiklohmann) [inline web/docs/operators/diagnostics.md:37]:
```suggestion
|`pipeline_id`|`string`|The ID of the pipeline that created the diagnostic.|
```

**Review by dominiklohmann — APPROVED** [review]
I created a follow-up story for adding a run id to metrics and diagnostics, other than that I think this can be shipped.

File: libtenzir/builtins/operators/diagnostics.cpp (hunk 1 above)
Comment (Dakostu) [reply libtenzir/builtins/operators/diagnostics.cpp]: Changed in https://github.com/tenzir/tenzir/pull/3828/commits/e25463acad78daf8b7adca97225cc3cbaa3fa14f

File: web/docs/operators/diagnostics.md (hunk 2 above)
Comment (Dakostu) [reply web/docs/operators/diagnostics.md:42]: Documented in https://github.com/tenzir/tenzir/pull/3828/commits/8e9c2cd861c547cb5fb8694fb44d02b70145f6e8

File: web/docs/operators/diagnostics.md (hunk 3 above)
Comment (Dakostu) [reply web/docs/operators/diagnostics.md]: Changed in https://github.com/tenzir/tenzir/pull/3828/commits/8e9c2cd861c547cb5fb8694fb44d02b70145f6e8

**Pull request merged**